File: Diagrammatize.Design.Pattern/code/cxx/Bridge/A2/FileDisplayImpl.cpp
#include "FileDisplayImpl.hpp"

#include <iostream>

FileDisplayImpl::FileDisplayImpl(const std::string& filename_): filename(filename_) {
}

void FileDisplayImpl::rawOpen() {
    reader.open(filename);
    std::cout << "=-=-=-=-=-= " << filename << " =-=-=-=-=-=" << std::endl; // 装饰框
}

void FileDisplayImpl::rawPrint() {
    std::string line;
    reader.clear();
    reader.seekg(0, reader.beg);
    while (std::getline(reader, line)) {
        std::cout << "> " << line << "\n";
    }
}

void FileDisplayImpl::rawClose() {
    std::cout << "=-=-=-=-=-= " << std::endl; // 装饰框
    reader.close();
}

File: Diagrammatize.Design.Pattern/code/python/Bridge/A2/FileDisplayImpl.py
from DisplayImpl import DisplayImpl

class FileDisplayImpl(DisplayImpl):
    def __init__(self, filename):
        self.filename = filename
        self.reader = None

    def rawOpen(self):
        self.reader = open(self.filename, "r")
        print("=-=-=-=-=-= " + self.filename + " =-=-=-=-=-=")

    def rawPrint(self):
        self.reader.seek(0, 0)
        for line in self.reader:
            print("> " + line)

    def rawClose(self):
        print("=-=-=-=-=-= ")
        self.reader.close()
